### PR TITLE
Adds Floorbot color variants

### PR DESCRIPTION
**Pull request by Arkatos1**
**What does this PR do:**
This PR adds differently colored Floorbot variants. Differently colored toolboxes will result in differently colored Floorbots. Floorbots made from Syndicate toolboxes will even have double health compared to the standard Floorbot. Also fixes a display bug in the crafting tab for the Floorbot, as required amount of floortiles needed to make a Floorbot is 10, not 1.

Finally, minor refactors of the relevant Floorbot files.

Ported from /tg/station13.

Tested locally without any issue.

**Images of sprite/map changes (IF APPLICABLE):**
TeamFloor

**Changelog:**
:cl: Arkatos
add: Added Floorbot color variants. Different toolboxes will result in a differently colored Floorbots. One even might be a little stronger than others..
imageadd: Added sprites of Floorbot color variants
fix: Fixed incorrect amount of required floortiles for the Floorbot in the crafting tab
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
So just making sure, that green one is from...?

**Comment by Arkatos1** [comment]
That would be plain green(artistic) toolbox. We have that toolbox fully working, but it's loot content and it's maintenance spawn was not implemented yet, but I expect it to be so in the future.
I have blacklisted His Grace, if anyone is concerned.

**Comment by Fox-McCloud** [comment]
"What did you do after purchasing His Grace? Did you kill everyone on station with it?"

"nah man, I made a floorbot"

". . ."

**Comment by KasparoVy** [comment]
TFW he makes a floorbot that goes around autonomously and devours people

**Comment by taukausanake** [comment]
@KasparoVy I would be all for that. I just wonder what the downside should be to doing that. It won't be made I know that, but it's interesting to think about it

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 1]
```diff
@@ -210,30 +210,53 @@ var/robot_arm = /obj/item/robot_parts/l_arm
...
+								/obj/item/storage/toolbox/green,
+								/obj/item/storage/toolbox/syndicate,
+								/obj/item/storage/toolbox/fakesyndi)
+
 	if(!istype(T, /obj/item/stack/tile/plasteel))
 		..()
 		return
+	if(!is_type_in_list(src, allowed_toolbox))
+		return
+	if(src.type == /obj/item/storage/toolbox/green/memetic)
```
Comment (variableundefined) [inline code/modules/mob/living/simple_animal/bot/construction.dm]: Implicit src

Comment (Arkatos1) [reply code/modules/mob/living/simple_animal/bot/construction.dm]: Hopefully fixed, if I understood correctly.

**Pull request merged**